src/vehicle.cpp

```diff
@@ -2134,6 +2134,20 @@ bool vehicle::has_part( const std::string &flag, bool enabled ) const
     } );
 }
 
+bool vehicle::has_part( const tripoint &pos, const std::string &flag, bool enabled ) const
+{
+    for( const auto &e : parts ) {
+        if( e.precalc[0].x != pos.x - global_x() ||
+            e.precalc[0].y != pos.y - global_y() ) {
+            continue;
+        }
+        if( !e.removed && !e.is_broken() && ( !enabled || e.enabled ) && ( flag.empty() || e.info().has_flag( flag ) ) ) {
```

Comment:
The `flag.empty()` part is undocumented and also not a default argument, meaning it would have to be explicitly invoked with as `has_part( pos, "" )`.